src/dotty/tools/dotc/transform/Pickler.scala

```diff
@@ -68,7 +68,7 @@ class Pickler extends Phase {
   private def pickleSourcefile(pickler: TastyPickler, source: SourceFile): Unit = {
     val buf = new TastyBuffer(10)
     pickler.newSection("Sourcefile", buf)
-    buf.writeNat(pickler.nameBuffer.nameIndex(source.file.path).index)
+    buf.writeNat(pickler.nameBuffer.nameIndex(source.file.canonicalPath).index)
```

Comment:
This may leak private knowledge of folder structure.
Why is relative path not enough?